GCRCatalogs/register.py

```diff
@@ -176,6 +185,23 @@ def set_root_dir_by_site(self, site):
             site_string = ' '.join(_config_register.site_list)
             warnings.warn(f"Unknown site '{site}'.\nAvailable sites are: {site_string}\nroot_dir is unchanged")
 
+    def persist_root_dir(self, value=None):
+        """
+        Write root_dir to user config. By default write current session value
+        """
+        old = self._user_config_manager.get(self._ROOT_DIR_KEY)
+        to_write = value or self.root_dir
+        self._user_config_manager[self._ROOT_DIR_KEY] = os.path.abspath(to_write)
```

Comment:
`old` is no longer needed. Add one blank line between class methods. 

One question --- should we allow the `value` argument? And if so, should we also set `self.root_dir = value` when `value` is not `None`? Given the function name `persist_root_dir`, I feel that allowing one to set the config to a value that is different from current `self.root_dir` seems a bit confusing?

Reply:
Given how this function is used in `set_root_dir`, I would suggest removing the `value` argument.

Reply:
The behavior is consistent with `unpersist_root_dir`, which only modifies the user config but leaves the session value alone.  I don't see harm in leaving the `value` argument there, even if it's not currently used, but if we change `persist_root_dir` I think we also need to reconsider `unpersist_root_dir`

Reply:
I think `unpersist_root_dir`'s behavior is expected as the function name suggests --- i.e., keep the current running session intact but remove the persistent effect, and it does not result in any conflicting settings. 

When `persist_root_dir` is called without `value`, its behavior is also expected (i.e., write current `root_dir` to config). But when `value` is supplied, I would expect its behavior to be similar to when `value` is not set (i.e, write current `root_dir` to config), and hence would expect it to first set current `root_dir`. If it doesn't, the the current session's `root_dir` would be in conflict with the value in config, which seems an odd situation. 

In particular, since `persist_root_dir` is a method of `RootDirManager` --- what is the use case when the user would want to use `RootDirManager` to set config but not the current session? 

If there is a clear use case, I would suggest making a different method for it, rather than overloading `persist_root_dir`.

Reply:
> I think unpersist_root_dir's behavior is expected as the function name suggests --- i.e., keep the current running session intact but remove the persistent effect, and it does not result in any conflicting settings.

I would say that is a conflict:  the persisted value is None and the session value isn't.  But as a practical matter for now, the `value` argument is unused so I will make the change.  We can revisit it when and if there is a need for it.